File: ClaimRuler/Lib/SchedulerNet/Samples/Scheduler.MVC3.VB/Controllers/SkinLocaleController.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc
Imports DHTMLX.Scheduler
Imports DHTMLX.Scheduler.Data
Imports DHTMLX.Scheduler.Controls
Imports DHTMLX.Common


Namespace SchedulerTest.Controllers

	''' <summary>
	''' check available skins and localizations
	''' </summary>
	Public Class SkinLocaleController
		Inherits Controller
		'
		' GET: /SkinLocale/
		Public Class LocaleData
			Public scheduler As DHXScheduler
			Public locale As String
			Public skin As String
			Public Sub New(sched As DHXScheduler, loc As String, sk As String)
				scheduler = sched
				locale = loc
				skin = sk
			End Sub
		End Class
		Public Function Index() As ActionResult
			Dim lang As SchedulerLocalization.Localizations
			Dim language = Me.Request.QueryString("language")
			Dim skn = Me.Request.QueryString("skin")
			Dim skin As DHXScheduler.Skins
			'#Region "language"
			Select Case Me.Request.QueryString("language")
				Case "ar"
					lang = SchedulerLocalization.Localizations.Arabic
					Exit Select
				Case "be"
					lang = SchedulerLocalization.Localizations.Belarusian
					Exit Select
				Case "ca"
					lang = SchedulerLocalization.Localizations.Catalan
					Exit Select
				Case "cn"
					lang = SchedulerLocalization.Localizations.Chinese
					Exit Select
				Case "cs"
					lang = SchedulerLocalization.Localizations.Czech
					Exit Select
				Case "da"
					lang = SchedulerLocalization.Localizations.Danish
					Exit Select
				Case "nl"
					lang = SchedulerLocalization.Localizations.Dutch
					Exit Select
				Case "fi"
					lang = SchedulerLocalization.Localizations.Finnish
					Exit Select
				Case "fr"
					lang = SchedulerLocalization.Localizations.French
					Exit Select
				Case "de"
					lang = SchedulerLocalization.Localizations.German
					Exit Select
				Case "el"
					lang = SchedulerLocalization.Localizations.Greek
					Exit Select
				Case "he"
					lang = SchedulerLocalization.Localizations.Hebrew
					Exit Select
				Case "hu"
					lang = SchedulerLocalization.Localizations.Hungarian
					Exit Select
				Case "id"
					lang = SchedulerLocalization.Localizations.Indonesia
					Exit Select
				Case "it"
					lang = SchedulerLocalization.Localizations.Italian
					Exit Select
				Case "jp"
					lang = SchedulerLocalization.Localizations.Japanese
					Exit Select
				Case "no"
					lang = SchedulerLocalization.Localizations.Norwegian
					Exit Select
				Case "pl"
					lang = SchedulerLocalization.Localizations.Polish
					Exit Select
				Case "pt"
					lang = SchedulerLocalization.Localizations.Portuguese
					Exit Select
				Case "ru"
					lang = SchedulerLocalization.Localizations.Russian
					Exit Select
				Case "ro"
					lang = SchedulerLocalization.Localizations.Romanian
					Exit Select
				Case "si"
					lang = SchedulerLocalization.Localizations.Slovenian
					Exit Select
				Case "es"
					lang = SchedulerLocalization.Localizations.Spanish
					Exit Select
				Case "sv"
					lang = SchedulerLocalization.Localizations.Swedish
					Exit Select
				Case "tr"
					lang = SchedulerLocalization.Localizations.Turkish
					Exit Select
				Case "ua"
					lang = SchedulerLocalization.Localizations.Ukrainian
					Exit Select
				Case Else
					lang = SchedulerLocalization.Localizations.English
					language = "en"
					Exit Select
			End Select
			'#End Region

			'#Region "skin"
			Select Case Me.Request.QueryString("skin")
				Case "glossy"
					skin = DHXScheduler.Skins.Glossy
					Exit Select
				Case "terrace"
					skin = DHXScheduler.Skins.Terrace
					Exit Select
				Case Else
					skin = DHXScheduler.Skins.Standart
					skn = "classic"
					Exit Select
			End Select
			'#End Region
			Dim scheduler = New DHXScheduler(Me)

			scheduler.InitialDate = New DateTime(2011, 11, 24)

			scheduler.XY.scroll_width = 0
			scheduler.Config.first_hour = 8
			scheduler.Config.last_hour = 19
			scheduler.Config.time_step = 30
			scheduler.Config.multi_day = True
			scheduler.Config.limit_time_select = True
			scheduler.Skin = skin
			scheduler.Localization.Directory = "locale"
			scheduler.Localization.[Set](lang, False)

			Dim rooms = New DHXSchedulerDataContext().Rooms.ToList()

			Dim unit = New UnitsView("unit1", "room_id")
			unit.AddOptions(rooms)
			'parse model objects
			scheduler.Views.Add(unit)

			Dim timeline = New TimelineView("timeline", "room_id")
			timeline.RenderMode = TimelineView.RenderModes.Bar
			timeline.FitEvents = False
			timeline.AddOptions(rooms)
			scheduler.Views.Add(timeline)


			scheduler.EnableDataprocessor = True
			scheduler.LoadData = True
			scheduler.InitialDate = New DateTime(2011, 9, 19)
			Return View(New LocaleData(scheduler, language, skn))
		End Function
		Public Function Data() As ContentResult

			Dim data__1 = New SchedulerAjaxData((New DHXSchedulerDataContext()).Events)


			Return data__1
		End Function

		Public Function Save(id As System.Nullable(Of Integer), actionValues As FormCollection) As ContentResult

			Dim action = New DataAction(actionValues)

			Dim data As New DHXSchedulerDataContext()
			Dim changedEvent = DirectCast(DHXEventsHelper.Bind(GetType([Event]), actionValues), [Event])
			Try
				Select Case action.Type
					Case DataActionTypes.Insert
						data.Events.InsertOnSubmit(changedEvent)
						Exit Select
					Case DataActionTypes.Delete
						changedEvent = data.Events.SingleOrDefault(Function(ev) ev.id = action.SourceId)
						data.Events.DeleteOnSubmit(changedEvent)
						Exit Select
					Case Else
						' "update"                          
						Dim eventToUpdate = data.Events.SingleOrDefault(Function(ev) ev.id = action.SourceId)
						DHXEventsHelper.Update(eventToUpdate, changedEvent, New List(Of String)() From { _
							"id" _
						})
						Exit Select
				End Select
				data.SubmitChanges()
				action.TargetId = changedEvent.id
			Catch a As Exception
				action.Type = DataActionTypes.[Error]
			End Try

			Return (New AjaxSaveResponse(action))
		End Function
	End Class
End Namespace

File: ClaimRuler/Lib/SchedulerNet/Samples/Scheduler.MVC4/Controllers/SkinLocaleController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using DHTMLX.Scheduler;
using DHTMLX.Scheduler.Data;
using DHTMLX.Scheduler.Controls;
using DHTMLX.Common;
using DHXScheduler_MVC4.Models;

namespace SchedulerTest.Controllers
{

    /// <summary>
    /// check available skins and localizations
    /// </summary>
    public class SkinLocaleController : Controller
    {
        //
        // GET: /SkinLocale/
        public class LocaleData
        {
            public DHXScheduler scheduler;
            public string locale;
            public string skin;
            public LocaleData(DHXScheduler sched, string loc, string sk)
            {
                scheduler = sched;
                locale = loc;
                skin = sk;
            }
        }
        public ActionResult Index()
        {
            SchedulerLocalization.Localizations lang;
            var language = this.Request.QueryString["language"];
            var skn = this.Request.QueryString["skin"];
            DHXScheduler.Skins skin;
            #region language
            switch (this.Request.QueryString["language"])
            {
                case "ar":
                    lang = SchedulerLocalization.Localizations.Arabic;
                    break;
                case "be":
                    lang = SchedulerLocalization.Localizations.Belarusian;
                    break;
                case "ca":
                    lang = SchedulerLocalization.Localizations.Catalan;
                    break;
                case "cn":
                    lang = SchedulerLocalization.Localizations.Chinese;
                    break;
                case "cs":
                    lang = SchedulerLocalization.Localizations.Czech;
                    break;
                case "da":
                    lang = SchedulerLocalization.Localizations.Danish;
                    break;
                case "nl":
                    lang = SchedulerLocalization.Localizations.Dutch;
                    break;
                case "fi":
                    lang = SchedulerLocalization.Localizations.Finnish;
                    break;
                case "fr":
                    lang = SchedulerLocalization.Localizations.French;
                    break;
                case "de":
                    lang = SchedulerLocalization.Localizations.German;
                    break;
                case "el":
                    lang = SchedulerLocalization.Localizations.Greek;
                    break;
                case "he":
                    lang = SchedulerLocalization.Localizations.Hebrew;
                    break;
                case "hu":
                    lang = SchedulerLocalization.Localizations.Hungarian;
                    break;
                case "id":
                    lang = SchedulerLocalization.Localizations.Indonesia;
                    break;
                case "it":
                    lang = SchedulerLocalization.Localizations.Italian;
                    break;
                case "jp":
                    lang = SchedulerLocalization.Localizations.Japanese;
                    break;
                case "no":
                    lang = SchedulerLocalization.Localizations.Norwegian;
                    break;
                case "pl":
                    lang = SchedulerLocalization.Localizations.Polish;
                    break;
                case "pt":
                    lang = SchedulerLocalization.Localizations.Portuguese;
                    break;
                case "ro":
                    lang = SchedulerLocalization.Localizations.Romanian;
                    break;
                case "ru":
                    lang = SchedulerLocalization.Localizations.Russian;
                    break;
                case "si":
                    lang = SchedulerLocalization.Localizations.Slovenian;
                    break;
                case "es":
                    lang = SchedulerLocalization.Localizations.Spanish;
                    break;
                case "sv":
                    lang = SchedulerLocalization.Localizations.Swedish;
                    break;
                case "tr":
                    lang = SchedulerLocalization.Localizations.Turkish;
                    break;
                case "ua":
                    lang = SchedulerLocalization.Localizations.Ukrainian;
                    break;
                default:
                    lang = SchedulerLocalization.Localizations.English;
                    language = "en";
                    break;
            }
            #endregion

            #region skin
            switch (this.Request.QueryString["skin"])
            {
                case "glossy":
                    skin = DHXScheduler.Skins.Glossy;
                    break;
                case "terrace":
                    skin = DHXScheduler.Skins.Terrace;
                    break;
                default:
                    skin = DHXScheduler.Skins.Standart;
                    skn = "classic";
                    break;
            }
            #endregion
            var scheduler = new DHXScheduler(this);
            scheduler.Skin = DHXScheduler.Skins.Terrace;
            scheduler.InitialDate = new DateTime(2011, 11, 24);

            scheduler.XY.scroll_width = 0;
            scheduler.Config.first_hour = 8;
            scheduler.Config.last_hour = 19;
            scheduler.Config.time_step = 30;
            scheduler.Config.multi_day = true;
            scheduler.Config.limit_time_select = true;
            scheduler.Skin = skin;
            scheduler.Localization.Directory = "locale";
            scheduler.Localization.Set(lang, false);

            var rooms = new DHXSchedulerModelsDataContext().Rooms.ToList();

            var unit = new UnitsView("unit1", "room_id");
            unit.AddOptions(rooms);//parse model objects
            scheduler.Views.Add(unit);

            var timeline = new TimelineView("timeline", "room_id");
            timeline.RenderMode = TimelineView.RenderModes.Bar;
            timeline.FitEvents = false;
            timeline.AddOptions(rooms);
            scheduler.Views.Add(timeline);


            scheduler.EnableDataprocessor = true;
            scheduler.LoadData = true;
            scheduler.InitialDate = new DateTime(2011, 9, 19);
            return View(new LocaleData(scheduler, language, skn));
        }
        public ContentResult Data()
        {

            var data = new SchedulerAjaxData((new DHXSchedulerModelsDataContext()).Events);
    

            return data;
        }

        public ContentResult Save(int? id, FormCollection actionValues)
        {

            var action = new DataAction(actionValues);

            DHXSchedulerModelsDataContext data = new DHXSchedulerModelsDataContext();
            var changedEvent = (Event)DHXEventsHelper.Bind(typeof(Event), actionValues);
            try
            {
                switch (action.Type)
                {
                    case DataActionTypes.Insert:
                        data.Events.InsertOnSubmit(changedEvent);
                        break;
                    case DataActionTypes.Delete:
                        changedEvent = data.Events.SingleOrDefault(ev => ev.id == action.SourceId);
                        data.Events.DeleteOnSubmit(changedEvent);
                        break;
                    default:// "update"                          
                        var eventToUpdate = data.Events.SingleOrDefault(ev => ev.id == action.SourceId);
                        DHXEventsHelper.Update(eventToUpdate, changedEvent, new List<string>() { "id" });
                        break;
                }
                data.SubmitChanges();
                action.TargetId = changedEvent.id;
            }
            catch
            {
                action.Type = DataActionTypes.Error;
            }

            return (new AjaxSaveResponse(action));
        }
    }
}
